tests/test_filters.py

```diff
@@ -871,6 +872,51 @@ def test_decimate_boundary():
     assert boundary.n_points
 
 
+def test_extract_surface():
+    # create a single hexahedral cell
+    lin_pts = np.array([[-1, -1, -1], # node 0
+                        [ 1, -1, -1], # node 1
+                        [ 1,  1, -1], # node 2
+                        [-1,  1, -1], # node 3
+                        [-1, -1,  1], # node 4
+                        [ 1, -1,  1], # node 5
+                        [ 1,  1,  1], # node 6
+                        [-1,  1,  1]], np.double) # node 7
+
+    quad_pts = np.array([
+        (lin_pts[1] + lin_pts[0])/2.0,
+        (lin_pts[1] + lin_pts[2])/2.0,
+        (lin_pts[2] + lin_pts[3])/2.0,
+        (lin_pts[3] + lin_pts[0])/2.0,
+        (lin_pts[4] + lin_pts[5])/2.0,
+        (lin_pts[5] + lin_pts[6])/2.0,
+        (lin_pts[6] + lin_pts[7])/2.0,
+        (lin_pts[7] + lin_pts[4])/2.0,
+        (lin_pts[0] + lin_pts[4])/2.0,
+        (lin_pts[1] + lin_pts[5])/2.0,
+        (lin_pts[2] + lin_pts[6])/2.0,
+        (lin_pts[3] + lin_pts[7])/2.0], np.double)
+
+    # introduce a minor variation to the location of the mid-side points
+    quad_pts += np.random.random(quad_pts.shape)*0.25
```

Comment:
It's probably not recommended to have random numbers in a test, but I can't think of an edge case where this could be problematic (relative shifts are pretty small).

It would also be nice to use the new random API (added in numpy 1.17 I think) for modernness' sake, but again that's not possible due to legacy versions.

Reply:
We use `hypothesis` quite a bit, and haven't had issues with random values.  I was honestly just being lazy here, but I think it works better than including an array.